repo.c

```diff
@@ -818,6 +818,8 @@ add_platform_module (const char *platform,
   modulemd_module_stream_set_context (module, "00000000");
   modulemd_module_stream_v2_set_arch ((ModulemdModuleStreamV2 *) module, arch);
   add_module_solvables (system, module);
+  /* Required for adding defaults */
+  pool_createwhatprovides (system->pool);
```

Comment:
as wrote in the issue, I think it should be in the different function than here.

Reply:
Should it be called on `pool` or `system->pool`? Both seem to work, but I would expect one to be better.

Reply:
In `create_system_repo` you can call it with `pool` directly imho.